baybe/acquisition/base.py

```diff
@@ -80,10 +86,33 @@ def to_botorch(
             additional_params["mc_points"] = to_tensor(
                 self.get_integration_points(searchspace)  # type: ignore[attr-defined]
             )
-
         params_dict.update(additional_params)
 
-        return acqf_cls(**params_dict)
+        acqf = acqf_cls(**params_dict)
+
+        if isinstance(self, qThompsonSampling):
+            assert hasattr(acqf, "_default_sample_shape")
```

Comment:
This is a workaround due to a current limitation in botorch to set the sample shape via the constructor (@julianStreibel has also opened a botorch issue for it). Because we cannot pass it, we set it after the object creation. The `hasattr` is just a sanity check that the property exists in the first case, just to make that part of the code future-proof as long as we keep it. Because `_default_sample_shape` is private, you must expect that botorch simply removes/renames this attribute. And it that case, our code would create a silent bug because you'd effectively create a new object attribute (slots send their warmest greetings). This way, we avoid that scenario

Reply:
Makes sense